### PR TITLE
B2FIND fix gui temporal search

### PR DESCRIPTION
**Pull request by juholehtonen**
UI fixes + in the last commit removed debug logs from getting into master.

### CODE REVIEW

File: ckanext/timeline/public/fanstatic/hc.js [hunk 1]
```diff
@@ -394,6 +395,37 @@ $(function () {
             else if (format == 'zero') {
                 c = Number(val);
             }
+            else if (format == 'dt' || format == 'dt2zero') {
+                var m = 0;
+                if (moment.utc(val.trim(), moment.ISO_8601, true).isValid()) {
+                    m = moment.utc(val.trim(), moment.ISO_8601, true);
+                    /* Pick the end of year for end boxes. */
+                    if (jquery.is('#end, #ext_timeline_end')) {
```
Comment (noumar) [inline ckanext/timeline/public/fanstatic/hc.js]: Don't use IDs directly. Use the end_box and end_box_hidden variables, if possible.

File: ckanext/timeline/public/fanstatic/hc.js [hunk 2]
```diff
@@ -394,6 +395,37 @@ $(function () {
...
+                        var tr = m.parsingFlags().parsedDateParts.length;
+                        m.endOf(timeResolution[tr-1]);
+                    }
+                }
+                else if (!isNaN(parseInt(val))) {
+                    m = moment.utc({'year': parseInt(val)});
+                    /* Pick the end of year for end boxes. */
+                    if (jquery.is('#end, #ext_timeline_end')) {
```
Comment (noumar) [inline ckanext/timeline/public/fanstatic/hc.js]: Same here.

File: ckanext/timeline/public/resource.config [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[groups]
+timeline =
+    fanstatic/moment.js
+    fanstatic/highcharts.js
+    fanstatic/hc.js
+
```
Comment (noumar) [inline ckanext/timeline/public/resource.config]: Can you remove the empty line?

File: ckanext/timeline/public/fanstatic/hc.js [hunk 4]
```diff
@@ -394,6 +395,37 @@ $(function () {
...
+                if (m !== 0) {
+                    if (format == 'dt') {
+                        c = m.format("YYYY-MM-DD HH:mm:ss")
+                    } else if (format == 'dt2zero') {
+                        c = helpers.datetimeAsZeroBased(m)
+                    }
+                }
+            }
+            else if (format == 'zero2dt'){
+                c = helpers.zeroBasedAsDatetime(val)
```
Comment (noumar) [inline ckanext/timeline/public/fanstatic/hc.js]: Semicolon

File: ckanext/timeline/public/fanstatic/hc.js [hunk 5]
```diff
@@ -394,6 +395,37 @@ $(function () {
...
+                    /* Pick the end of year for end boxes. */
+                    if (jquery.is('#end, #ext_timeline_end')) {
+                        m.endOf('year');
+                    }
+                }
+                if (m !== 0) {
+                    if (format == 'dt') {
+                        c = m.format("YYYY-MM-DD HH:mm:ss")
+                    } else if (format == 'dt2zero') {
+                        c = helpers.datetimeAsZeroBased(m)
```
Comment (noumar) [inline ckanext/timeline/public/fanstatic/hc.js]: Semicolon

File: ckanext/timeline/public/fanstatic/hc.js [hunk 6]
```diff
@@ -394,6 +395,37 @@ $(function () {
...
+                else if (!isNaN(parseInt(val))) {
+                    m = moment.utc({'year': parseInt(val)});
+                    /* Pick the end of year for end boxes. */
+                    if (jquery.is('#end, #ext_timeline_end')) {
+                        m.endOf('year');
+                    }
+                }
+                if (m !== 0) {
+                    if (format == 'dt') {
+                        c = m.format("YYYY-MM-DD HH:mm:ss")
```
Comment (noumar) [inline ckanext/timeline/public/fanstatic/hc.js]: Semicolon

**Review by noumar — CHANGES_REQUESTED** [review]

File: ckanext/timeline/public/fanstatic/hc.js (hunk 1 above)
Comment (juholehtonen) [reply ckanext/timeline/public/fanstatic/hc.js]: Ok. Changed.

File: ckanext/timeline/public/fanstatic/hc.js (hunk 2 above)
Comment (juholehtonen) [reply ckanext/timeline/public/fanstatic/hc.js]: Ok.

File: ckanext/timeline/public/fanstatic/hc.js (hunk 6 above)
Comment (juholehtonen) [reply ckanext/timeline/public/fanstatic/hc.js]: Ok.

File: ckanext/timeline/public/fanstatic/hc.js (hunk 5 above)
Comment (juholehtonen) [reply ckanext/timeline/public/fanstatic/hc.js]: Ok.

File: ckanext/timeline/public/fanstatic/hc.js (hunk 4 above)
Comment (juholehtonen) [reply ckanext/timeline/public/fanstatic/hc.js]: Ok.

File: ckanext/timeline/public/resource.config (hunk 3 above)
Comment (juholehtonen) [reply ckanext/timeline/public/resource.config]: Removed

**Comment by noumar** [comment]
@juholehtonen will make some small changes. Can @mkurtz then deploy this to 6c or 6d? Once you are satisfied @mkurtz can merge the PR.

**Review by noumar — APPROVED** [review]

**Pull request merged**